server/chat-plugins/scavengers.js

```diff
@@ -216,6 +222,73 @@ function formatOrder(place) {
 	return place + 'th';
 }
 
+class ScavengerHuntDatabase {
+	static getRecycledHuntFromDatabase() {
+		// Return a random hunt from the database.
+		return scavengersData.recycledHunts[Math.floor(Math.random() * scavengersData.recycledHunts.length)];
+	}
+
+	static addRecycledHuntToDatabase(hosts, params) {
+		if (!scavengersData.recycledHunts) {
+			scavengersData.recycledHunts = [];
+		}
+
+
+		const huntSchema = {
+			hosts: hosts,
+			questions: [],
+			get fullText() {
+				return `${this.hosts.map(host => host.name).join(',')} | ${this.questions.map(question => `${question.text} | ${question.answers.join(';')}`).join(' | ')}`;
+			},
+		};
+
+		let questionSchema = {
+			text: '',
+			answers: [],
+			hints: [],
+		};
+
+		for (let i = 0; i < params.length; ++i) {
+			if (i % 2 === 0) {
+				questionSchema.text = params[i];
+			} else {
+				questionSchema.answers = params[i];
+				huntSchema.questions.push(questionSchema);
+				questionSchema = {
+					text: '',
+					answers: [],
+					hints: [],
+				};
+			}
+		}
+
+		scavengersData.recycledHunts.push(huntSchema);
+		FS(DATABASE_FILE).writeUpdate(() => JSON.stringify(scavengersData));
+	}
+
+	static removeRecycledHuntFromDatabase(index) {
+		scavengersData.recycledHunts.splice(index - 1, 1);
+		FS(DATABASE_FILE).writeUpdate(() => JSON.stringify(scavengersData));
+	}
+
+	static addHintToRecycledHunt(huntNumber, questionNumber, hint) {
+		scavengersData.recycledHunts[huntNumber - 1].questions[questionNumber - 1].hints.push(hint);
+		FS(DATABASE_FILE).writeUpdate(() => JSON.stringify(scavengersData));
+	}
+
+	static removeHintToRecycledHunt(huntNumber, questionNumber, hintNumber) {
+		scavengersData.recycledHunts[huntNumber - 1].questions[questionNumber - 1].hints.splice(hintNumber - 1);
+		FS(DATABASE_FILE).writeUpdate(() => JSON.stringify(scavengersData));
+	}
+
+	static isEmpty() {
+		return scavengersData && scavengersData.recycledHunts && scavengersData.recycledHunts.length !== 0;
```

Comment:
Get rid of those unnecessary null checks. `scavengersData` will always be non-null, and `scavengersData.recycledHunts` should be properly initialized to an empty array (it's not currently, but that would be a good idea to avoid bugs).

Reply:
Now intializing `scavengersData.recycledHunts` to an empty array.